hazelcast/src/main/java/com/hazelcast/spi/utils/RestClient.java

```diff
@@ -168,6 +172,47 @@ private Response call(String method) {
         }
     }
 
+    /**
+     * Issues a watch request to a Kubernetes resource, starting with the given {@code resourceVersion}.
+     * Since a watch implies a stream of updates from the server will be consumed, unlike other methods
+     * in this class, it is the responsibility of the consumer to disconnect the connection
+     * (by invoking {@link WatchResponse#disconnect()}) once the watch is no longer required.
+     */
```

Comment:
The upcoming question is not related to the PR 🙂 

What if we want to convert the existing discovery mechanism into a mechanism like this one, a more dynamic version?  In the current version, the latest member discovers existing ones via running related REST call-based methods.